sdk/core/azure-core/src/http/curl/curl.cpp

```diff
@@ -1012,10 +1012,51 @@ std::map<std::string, std::list<std::unique_ptr<CurlNetworkConnection>>>
 int32_t CurlConnectionPool::s_connectionCounter = 0;
 bool CurlConnectionPool::s_isCleanConnectionsRunning = false;
 
+namespace {
+inline std::string GetConnectionPropertiesKey(CurlTransportOptions const& options)
```

Comment:
These connection string keys **must** be unique and map exactly to 1 option setting with **no collisions**. Come up with an algorithm that guarantees that and is easy to audit and test. There should always be a 1:1 mapping between key string and options (i.e. if I have a key, I can figure out which unique and exactly one option settings it corresponds to, and if I have an option, exactly one unique string can be generated from it).

Currently, two keys that are identical might come from different option values. We need to be careful to avoid that.

Here's an example: "0111"
Now, tell me, what options correspond to that connection key? Is there only a single unique options setting that results in that key?

I am sure there are other possible such strings that cause a collision.

<details><summary>Try it for yourself first, but here's the answer:</summary>
Based on the cainfo and proxy setting strings, I can spoof it and get a connection where keep alive to be on.

0<1>11 <= cainfo is "", keep alive is 0, proxy is "1", norevoke is 1, sslverifypeer is 1
<0>111 <= cainfo is "0", keep alive is 1, proxy is "", norevoke is 1, sslverifypeer is 1
</details>

Reply:
If you are setting CAPath to "0", you are going to have other issues XD.
But I can fix it for sure :)